file.go

```diff
@@ -221,6 +216,63 @@ type Undo struct {
 	buf []rune
 }
 
+func (f *File) InsertAt(p0 int, s []rune) {
+	f.treatasclean = false
+	if p0 > f.b.nc() {
+		panic("internal error: fileinsert")
+	}
+	if f.seq > 0 {
+		f.Uninsert(&f.delta, p0, len(s))
+	}
+	f.b.Insert(p0, s)
+	if len(s) != 0 {
+		f.Modded()
+	}
+	f.AllObservers(func(i interface{}) {
+		i.(BufferObserver).inserted(p0, s)
+	})
+}
+
+func (f *File) InsertAtWithoutCommit(p0 int, s []rune) {
+	f.treatasclean = false
+	if p0 > f.b.nc()+len(f.cache) {
+		panic("File.InsertAtWithoutCommit insertion off the end")
+	}
+
+	if len(f.cache) == 0 {
+		f.cq0 = p0
+	} else {
+		if p0 != f.cq0+len(f.cache) {
+			// TODO(rjk): actually print something useful here
+			acmeerror("File.InsertAtWithoutCommit cq0", nil)
```

Comment:
Is this what you mean by "the warnings"?

Reply:
Yes